src/components/calendar/drag-drop-provider.tsx

```
@@ -0,0 +1,353 @@
+  listRef: React.RefObject<any>,
+  currentScrollOffsetRef: React.RefObject<number>,
+  viewportHeightRef: React.RefObject<number>
+) {
+  const scrollBy = React.useCallback(
+    (delta: number) => {
+      const ref = listRef.current as any;
+      if (!ref || typeof (ref as any).scrollToOffset !== 'function') return;
+      const current = currentScrollOffsetRef.current;
+      const next = Math.max(0, current + delta);
+      (ref as any).scrollToOffset({ offset: next, animated: false });
+      currentScrollOffsetRef.current = next;
```

ESLint

[error] Mutating component props or hook arguments is not allowed. Consider using a local variable instead

(react-compiler/react-compiler)